crates/ln-dlc-node/src/ln/event_handler.rs

```diff
@@ -997,3 +792,17 @@ pub mod handlers {
         });
     }
 }
+#[async_trait]
+impl<T: EventHandlerTrait + ?Sized> EventHandlerTrait for Arc<T> {
+    async fn match_event(&self, event: Event) -> Result<()> {
+        (**self).match_event(event).await
+    }
+
+    async fn handle_event(&self, event: Event) {
+        (**self).handle_event(event).await
+    }
```

Comment:
❓Why do we need two methods?

Reply:
it became evident in the following commit, where I made sure that every event handler has the option to send the events (to avoid reimplementing that.
`handle_event()` has a default impl that's shared, and the implementers only need to implement:

- `event_sender() -> &EventSender` - how to retrieve `EventSender` 
- `match_event()` - how to implement actual event logic.